tests/hijack/db.js

```diff
@@ -1,534 +1,514 @@
-      ['db',undefined,{coll: 'tinytest-data', cursor: true, func: 'forEach', selector: JSON.stringify({_id: {$exists: true}})}],
+      ['start',{userId: null, params: '[]'}],
+      ['wait',{waitOn: []}],
+      ['db',{coll: 'tinytest-data', cursor: true, func: 'forEach', selector: JSON.stringify({_id: {$exists: true}})}],
       ['complete']
     ];
-    test.equal(result, ['aa', 'bb']);
-    test.equal(events, expected);
-    CleanTestData();
+
+    test.stableEqual(result, ['aa', 'bb']);
+    test.stableEqual(events, expected);
   }
 );
 
-Tinytest.add(
+addAsyncTest(
   'Database - Cursor - observeChanges',
-  function (test) {
-    EnableTrackingMethods();
-    TestData.insert({_id: 'aa'});
-    TestData.insert({_id: 'bb'});
-    let methodId = RegisterMethod(function () {
+  async function (test) {
+    await TestData.insertAsync({_id: 'aa'});
+    await TestData.insertAsync({_id: 'bb'});
+
+    let methodId = registerMethod(async function () {
       let data = [];
-      let handle = TestData.find({}).observeChanges({
+
+      let handle = await TestData.find({}).observeChanges({
         added (id, fields) {
           fields._id = id;
           data.push(fields);
         }
       });
+
       handle.stop();
+
       return data;
     });
-    let client = GetMeteorClient();
-    let result = client.call(methodId);
-    let events = GetLastMethodEvents([0, 2]);
-    events[3][2].oplog = false;
+
+    let result = await callAsync(methodId);
+
+    let events = getLastMethodEvents([0, 2], ['noOfCachedDocs']);
+
+    events[2][1].oplog = false;
+
     let expected = [
-      ['start',undefined,{userId: null, params: '[]'}],
-      ['wait',undefined,{waitOn: []}],
-      ['db',undefined,{coll: 'tinytest-data', func: 'find', selector: JSON.stringify({})}],
-      ['db',undefined,{coll: 'tinytest-data', cursor: true, func: 'observeChanges', selector: JSON.stringify({}), oplog: false, noOfCachedDocs: 2, wasMultiplexerReady: false}],
+      ['start',{userId: null, params: '[]'}],
+      ['wait',{waitOn: []}],
+      ['db',{coll: 'tinytest-data', cursor: true, func: 'observeChanges', selector: JSON.stringify({}), oplog: false, noOfCachedDocs: 2}],
       ['complete']
     ];
-    test.equal(result, [{_id: 'aa'}, {_id: 'bb'}]);
-    clearAdditionalObserverInfo(events[3][2]);
-    test.equal(events, expected);
-    CleanTestData();
+
+    test.stableEqual(result, [{_id: 'aa'}, {_id: 'bb'}]);
+
+    clearAdditionalObserverInfo(events[2][1]);
+
+    test.stableEqual(events, expected);
   }
 );
 
-Tinytest.add(
-  'Database - Cursor - observeChanges:re-using-multiflexer',
-  function (test) {
-    EnableTrackingMethods();
-    TestData.insert({_id: 'aa'});
-    TestData.insert({_id: 'bb'});
-    let methodId = RegisterMethod(function () {
+
+/**
+ * @warning `wasMultiplexerReady` is true for both when it should be false for the first one. Which might be an issue in Meteor code, so let's not test that.
```

Comment:
Is this specific to Meteor 3?

Reply:
Yes

Reply:
It seems to be flaky, sometimes it will be false and others true, our wrapper doesn't seem to be at fault.